src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractKeyedCodeElement.cs

```diff
@@ -106,7 +107,14 @@ protected override string GetName()
 
         protected override void SetName(string value)
         {
-            UpdateNodeAndReaquireNodeKey(FileCodeModel.UpdateName, value);
+            var nodeKeyValidation = new NodeKeyValidation();
```

Comment:
Great question! The answer is, yes! It's always better to be paranoid that the invisible editor is created.